libbeat/processors/decompress/gzip.go

```diff
@@ -0,0 +1,153 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package gzip
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/elastic/beats/libbeat/processors/checks"
+
+	"bytes"
+	"compress/gzip"
+	"io"
+)
+
+type decompressGzipFields struct {
+	decompressGzipFieldsConfig
+	fields map[string]string
+}
+
+type decompressGzipFieldsConfig struct {
+	Fields        common.MapStr `config:"fields"`
```

Comment:
We should not let users configure more fields for a single processor. This way if processing of a field fails, the source of the error is easier to find. No need to look for the field which caused the issue. This is a good example: https://github.com/elastic/beats/blob/master/libbeat/processors/actions/decode_base64_field.go